### PR TITLE
[WIP] whisper large fine-tuning on wenetspeech, mutli-hans-zh

### PR DESCRIPTION
**Pull request by yuekaizhang**
WenetSpeech (large-v2, 4 epochs, 3.3 days using 8*A100)

|     epoch                   |   Test_meeting (Greedy Search)       |   Errors                      
|------------------------|------|-------|
| pretrained (before fine-tuning)        | 24.84% | 3182 insertions, 36796 deletions, 14758 substitutions | 
|         0.067 epoch                              | 16.19% | 2087 insertions, 18459 deletions, 15143 substitutions |
|         0.2 epoch                                                 | 15.99% | 1886 insertions, 20185 deletions, 13166 substitutions |
|         0.4 epoch        | 16.29% | 1478 insertions, 22128 deletions, 12303 substitutions | 
|         0.53 epoch                              | 17.0% | 1235 insertions, 25044 deletions, 11184 substitutions |
|         1.0 epoch                                                 | 15.56% | 1707 insertions, 21817 deletions, 10763 substitutions |
|         2.27 epoch        | 16.73% | 1862 insertions, 25088 deletions, 9912 substitutions | 
|         3.53 epoch                              | 18.78% | 1215 insertions, 31504 deletions, 8673 substitutions |

Tensorboard: https://huggingface.co/yuekai/icefall_asr_wenetspeech_whisper/tensorboard

Error file: https://huggingface.co/yuekai/icefall_asr_wenetspeech_whisper/blob/main/exp_large_v2/errs-TEST_MEETING-beam-search-epoch-1-avg-1.txt

Top 10 Deletion Errors 14559 / 21817, (66.7% total deletion errors)
DELETIONS: count ref
3000   儿
2266   呃
2012   嗯
1814   个
1288   啊
1152   这
983   是
769   就
724   那
551   呢

I wonder if anyone has any ideas on how to reduce so many deletion errors? I'm willing to try if someone has any ideas.

### CODE REVIEW

**Comment by nshmyrev** [comment]
Not sure about deletion errors, but the key aspect of whisper is that it is multiobjective both transcription and translation. Translation task is not just for fun, it helps model to factorize out grammar and sense aspect and improve ASR accuracy.

If you want to get a good numbers on high-resource language, you also need multi-objective, at least translation task. Ideally speakerid objective too.

**Comment by csukuangfj** [comment]
Have you tried model average using --iter instead of --epoch?

**Comment by yuekaizhang** [comment]
> Have you tried model average using --iter instead of --epoch?

Not yet. Above results didn't use any model average techniques. Btw, since the model was wrapped by deepspeed, I can't use k2's model avg technique and do average after training manually.

**Comment by yuekaizhang** [comment]
> Not sure about deletion errors, but the key aspect of whisper is that it is multiobjective both transcription and translation. Translation task is not just for fun, it helps model to factorize out grammar and sense aspect and improve ASR accuracy.
> 
> If you want to get a good numbers on high-resource language, you also need multi-objective, at least translation task. Ideally speakerid objective too.

Good point. It would be great if you know some experiment results or papers which use multi-objective to fine-tune whisper.

**Comment by marcoyang1998** [comment]
Have other people reported similar observations when fine-tuning Whisper on WenetSpeech?

Similarly, we experienced severe deletion errors while training Zipformer on WenetSpeech (see #1130), could this be the problem of the dataset?

**Comment by yuekaizhang** [comment]
> Have other people reported similar observations when fine-tuning Whisper on WenetSpeech?
> 
> Similarly, we experienced severe deletion errors while training Zipformer on WenetSpeech (see #1130), could this be the problem of the dataset?

@marcoyang1998 You're correct, see https://github.com/wenet-e2e/WenetSpeech/discussions/54. 

One solution is to retrain with the new labels provided by https://github.com/wenet-e2e/WenetSpeech/discussions/54, but I'm thinking that for such colloquial scenarios, there might be a better way to evaluate, such as reducing the weight of errors related to modal particles. 

When people want to use ASR to add subtitles to their videos, it's clear that it would be more helpful if the model could automatically omit these colloquial words.

File: egs/speechio/ASR/zipformer/icefall-asr-multi-zh-hans-zipformer-ctc-2023-10-24
Comment (JinZr) [inline egs/speechio/ASR/zipformer/icefall-asr-multi-zh-hans-zipformer-ctc-2023-10-24:1]: is it necessary to include a submodule entry for this recipe?

**Comment by JinZr** [comment]
hi, thank you for your work!

i went through the pr and left a comment and few modifications, if you feel like those are proper then i think this pr is ready to merge.

File: egs/speechio/ASR/zipformer/icefall-asr-multi-zh-hans-zipformer-ctc-2023-10-24
Comment (yuekaizhang) [reply egs/speechio/ASR/zipformer/icefall-asr-multi-zh-hans-zipformer-ctc-2023-10-24:1]: Done. It was a mistake.

**Review by JinZr — APPROVED** [review]
LGTM, waiting for CI tests to be done

thanks!
